### ISSUE TITLE
Typescript typings incompatible with webpack 5.

### ISSUE THREAD

**Issue by wegry** [shellscape/webpack-manifest-plugin#279]
- Manifest Plugin Version: 4.0.2
- Webpack Version: 5.64.1
- Operating System (or Browser): macOS
- Node Version: 16.13.0
- Link to reproduction _(⚠️ read below)_:

https://github.com/wegry/webpack-5typescript-types-breaking-webpack-manifest-plugin/tree/main

### Expected Behavior

webpack-manifest-plugin should compile with tsc with webpack@5.

### Actual Behavior

webpack-manifest-plugin seems to be using webpack 4 typings instead.

### Additional Information

Originally I ran into this issue in `yarn 3` with `nodeLinker: node-modules`. The repro repo is yarn 1 though.

**Comment by shellscape** [comment]
Yeah I doubt we'll be able to resolve that until we drop Webpack v4 support.

**Comment by SunHuawei** [comment]
I'm experiencing the same issue. So, is there any plan?

**Comment by shellscape** [comment]
@SunHuawei did you read my last reply?

**Comment by SunHuawei** [comment]
@shellscape , yes, I did. I mean is there a plan to drop v4 support? I know that's not easy for the existing user, however, due to that it's a breaking change, is that possible to just upgrade to v5 for this plugin to support webpack@5. That would be perfect for both v4 and v5 users.

**Comment by shellscape** [comment]
I don't currently have the motivation to make that change.

**Comment by SunHuawei** [comment]
I don't get it, do you mean you don't see any value in it? Or, you're too busy? Honestly, I love this project, and I need it very much. If you can help review and merge, I can spend some time doing that upgrade. What do you think?

**Comment by shellscape** [comment]
community contribution is always welcome

**Comment by SunHuawei** [comment]
I've submitted a PR here https://github.com/shellscape/webpack-manifest-plugin/pull/289.
